### DISCUSSION TITLE
What are the available optimization techniques for array type fields?

### DISCUSSION THREAD

**Discussion by xpzhu2025** [apache/doris#40405] [A - General / Q&A]
The business scenario involves migrating from Druid to Doris 2.1.0, where it was observed that the performance is lower than Druid, especially for array types. The duplicate model is being used, and all optimization methods mentioned in the documentation (such as materialized views and indexes) have been tried without success. The data volume is very large, amounting to several billion records.

example:
tags: ARRAY<VARCHAR(20)>
sql: SELECT ext.`tag` AS `tag`, COUNT(*) AS `count` FROM `tableName` AS t LATERAL VIEW EXPLODE(`tags`) ext AS `tag`  WHERE ('2024-09-04T06:30:00Z'<=(t.`datetime`) AND (t.`datetime`)<'2024-09-05T07:00:00Z') GROUP BY 1 ORDER BY `count` DESC

**Comment by ixzc** [comment]
建议英文提问，如使用中文提问，可以去ask.selectdb.com中文技术论坛。
